com.unity.ml-agents/Runtime/ITeamManager.cs

```diff
@@ -0,0 +1,14 @@
+using System.Collections.Generic;
+using Unity.MLAgents.Sensors;
+
+namespace Unity.MLAgents
+{
+    public interface ITeamManager
```

Comment:
ITeamManager holds reference to a list of Agent and Agent has a reference to iTeamManager. This is a cyclic reference I think we should avoid. 
We could get away with it if we made Agent implement an interface and have ITeamManager hold a list of objects that implements that new interface.

Reply:
It looks like the Agent doesn't really need the TeamManager that much. Maybe it could avoid storing the reference to it altogether; you'd mostly need to
* add an internal method on Agent to set the teamManagerId and have BaseTeamManager call it
* make sure Agent unregisters itself with the teamManager when it gets destroyed. I think we could handle this with a delegate on Agent.

Reply:
Removed TeamManager reference in agent and unregister agent using delegate.